src/perform_step/pdirk_perform_step.jl

```diff
@@ -11,38 +11,36 @@ end
   @unpack nlsolver,k1,k2 = cache
   if alg.threading == true
     Threads.@threads for i in 1:2
```

Comment:
you should put `let` blocks around this, otherwise you lose type-stability. It's a bug with the captured variables from some anonymous functions. See the other multithreaded methods for examples